experimental/segmenter/src/lstm_bies.rs

```diff
@@ -50,69 +49,46 @@ impl<'l> Lstm<'l> {
         }
 
         // TODO: Perform this matrix reshaping in datagen.
-        let mat1 = data.get().mat1.as_ndarray2()?;
-        let mat2 = data.get().mat2.as_ndarray2()?;
-        let mat3 = data.get().mat3.as_ndarray2()?;
-        let mat4 = data.get().mat4.as_ndarray1()?;
-        let mat5 = data.get().mat5.as_ndarray2()?;
-        let mat6 = data.get().mat6.as_ndarray2()?;
-        let mat7 = data.get().mat7.as_ndarray1()?;
-        let mat8 = data.get().mat8.as_ndarray2()?;
-        let mat9 = data.get().mat9.as_ndarray1()?;
-        let embedd_dim = *mat1.shape().get(1).ok_or(Error::DimensionMismatch)?;
-        let hunits = *mat3.shape().first().ok_or(Error::DimensionMismatch)?;
-        let backward_hunits = *mat6.shape().first().ok_or(Error::DimensionMismatch)?;
-        if mat2.shape() != [embedd_dim, 4 * hunits]
-            || mat3.shape() != [hunits, 4 * hunits]
-            || mat4.shape() != [4 * hunits]
-            || mat5.shape() != [embedd_dim, 4 * hunits]
-            || mat6.shape() != [hunits, 4 * hunits]
-            || mat7.shape() != [4 * hunits]
-            || mat8.shape() != [2 * hunits, 4]
-            || mat9.shape() != [4]
+        let mat1 = data.get().mat1.alloc_matrix::<2>()?;
```

Comment:
As discussed: I plan to benchmark whether we can read directly from the zerovec without degrating LSTM performance, but LSTM is already highly performance-sensitive so it may be the case that we need to eat this cost. I'll add a comment.